src/frontend/src/views/Upload/index.tsx

```diff
@@ -29,8 +30,18 @@ export default function Upload(): JSX.Element | null {
 
   const router = useRouter();
 
+  // To avoid possible race condition in Metadata upload where we need locations,
+  // we fetch locations info early in parent container, so ready by that step.
+  // FIXME (Vince): Need to put in place error handling if fetch fails.
+  // Right now the use of empty array prevents a hard crash, but user will be
+  // confused because page will seem fine but location finding just won't work.
+  const { data: namedLocationsData } = useNamedLocations();
+  const namedLocations =
+    namedLocationsData?.namedLocations || ([] as NamedGisaidLocation[]);
+
```

Comment:
I think this is fine. If the fetch for locations fails, I doubt uploading samples is going to work!

Reply:
For sure! I don't think we really need anything fancy, it would just be nice to have some kind of generic thing across the app for when one of our API calls fail, something like, "There has been an error, please refresh the page. If this persists, contact us." If any API call fails, I think pretty much everything is hosed, but right now failure is quiet and the app might get into weird places where it looks like it's functioning before glitching out later.